src/site/content/en/blog/progressively-enhance-your-pwa/index.md

```diff
@@ -0,0 +1,1052 @@
+
+```js
+if ('setAppBadge' in navigator) {
+  import('./badge.mjs');
+}
+```
+
+In this example, I have drawn the numbers from one to seven, using one pen stroke
+per number.
+The badge counter on the icon is now at seven.
+
+<figure class="w-figure">
+  <img class="w-screenshot"
+       src="10000201000009C4000005DBE9BF427ABD63947A.png"
+       alt="The numbers from one to seven drawn onto the greeting card, each with just one pen stroke.">
+  <figcaption class="w-figcaption">
+    Drawing the numbers from 1 to 7, using seven pen strokes.
+  </figcaption>
+</figure>
+
+<figure class="w-figure">
+  <img class="w-screenshot"
+       src="10000201000002E6000001C0855AA2DF1E30228C.png"
+       alt="Badge icon on the Fugu Greetings app showing the number 7."
+       width="300">
+  <figcaption class="w-figcaption">
+    The pen strokes counter in the form of the app icon badge.
+  </figcaption>
+</figure>
+
+## The Periodic Background Sync API
+
+Want to start each day fresh with something new?
+A neat feature of the Fugu Greetings app is that it can inspire you each morning
+with a new background image to start your greeting card.
+The app uses the [Periodic Background Sync API](https://web.dev/periodic-background-sync/)
+to achieve this.
+
+The first step is to *register* a periodic sync event in the service worker registration.
+It listens for a sync tag called `'image-of-the-day'`
+and has a minimum interval of one day,
+so the user can get a new background image every 24 hours.
+
+```js
+const registerPeriodicBackgroundSync = async () => {
+  const registration = await navigator.serviceWorker.ready;
+  try {
+    registration.periodicSync.register('image-of-the-day-sync', {
+      // An interval of one day.
+      minInterval: 24 * 60 * 60 * 1000,
+    });
+  } catch (err) {
+    console.error(err.name, err.message);
+  }
+};
+```
+
+The second step is to *listen* for the `periodicsync` event in the service worker.
+If the event tag is `'image-of-the-day'`, that is, the one that was registered before,
+the image of the day is retrieved via the `getImageOfTheDay()` function,
+and the result propagated to all clients, so they can update their canvases and
+caches.
+
+```js
+self.addEventListener('periodicsync', (syncEvent) => {
+  if (syncEvent.tag === 'image-of-the-day-sync') {
+    syncEvent.waitUntil(
+      (async () => {
+        const blob = await getImageOfTheDay();
+        const clients = await self.clients.matchAll();
+        clients.forEach((client) => {
+          client.postMessage({
+            image: blob,
+          });
+        });
+      })()
+    );
+  }
+});
+```
+
+Again this is truly a progressive enhancement, so the code is only loaded when the
+API is supported by the browser.
+This applies to both the client code and the service worker code.
+On non-supporting browsers, neither of them is loaded.
+Note how in the service worker, instead of a dynamic `import()`, I use the classic
```

Comment:
```suggestion
Note how in the service worker, instead of a dynamic `import()`
(that isn't supported in a service worker context yet), I use the classic
```